generate_changelog.py

```
@@ -0,0 +1,644 @@
+
+    if not prev_date or not curr_date:
+        # Fall back to git tag dates
+        if prev_tag:
+            prev_date = run(f'git log -1 --format=%ai "{prev_tag}"')
+        curr_date = run(f'git log -1 --format=%ai "{current_tag}"')
+
+    if not prev_date or not curr_date:
+        return []
+
+    raw = run(
+        f'gh pr list --state merged --base main --limit 500 '
```

Ruff (0.15.21)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)